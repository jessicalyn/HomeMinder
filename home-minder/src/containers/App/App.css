.app {
  text-align: center;
}

.app-header {
  align-items: center;
  justify-content: center;
}

.nav-link {
  padding: 1rem
}