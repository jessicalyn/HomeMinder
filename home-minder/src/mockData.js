export const mockHomeData = [
  { id: "JK1234",
    room: "Kitchen",
    category: "Filters",
    name: "Refrigerator Water Filter",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "6 months",
    userScheduled: "",
    type: "Replace"
  },
  { id: "IF1234",
    room: "Kitchen",
    category: "Sponges",
    name: "Sponge",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "2-4 weeks",
    userScheduled: "",
    type: "Replace"
  },
  { id: "PF123",
    room: "Bathroom",
    category: "Toothbrush",
    name: "Toothbrush",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "1-3 months",
    userScheduled: "",
    type: "Replace"
  },
]