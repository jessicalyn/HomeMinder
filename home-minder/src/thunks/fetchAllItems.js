import { storeAllItems, hasError, isLoading } from '../actions'

export const fetchAllItems = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const data = await response.json()
      dispatch(storeAllItems(data))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}