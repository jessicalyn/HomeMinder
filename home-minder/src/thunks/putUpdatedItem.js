import { isLoading, hasError, updateItem } from "../actions";

export const putUpdatedItem = (url, options) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, options)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      // dispatch(updateItem(result))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}