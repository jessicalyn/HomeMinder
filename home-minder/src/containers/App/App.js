import React, { Component } from 'react';
import { Switch, Route, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import './App.css';
import Audit from '../Audit/Audit';
import error404 from '../../components/error404/error404';
import HomeContainer from '../Home Container/HomeContainer';
import { fetchAllItems } from '../../thunks/fetchAllItems';
import HomeCard from '../HomeCard/HomeCard'

export class App extends Component {

  componentDidMount() {
    const url = "http://localhost:3001/api/v1/home"
    this.props.fetchAllItems(url)
  }

  render() {
    const { error, items } = this.props
    return (
      <div className='app'>
        <header className='app-header'>
          <h1>HomeMinder</h1>
          <NavLink className="nav-link" to='/'>Home</NavLink>
          <NavLink className="nav-link" to='/audit'>Audit</NavLink>
          { error }
        </header>
        <div className='app-container'>
          <Switch>
            <Route exact path='/' component={HomeContainer}/>
            <Route path="/audit" component={Audit}/>
            <Route path='/item/:id' render={({ match }) => {
              const { id } = match.params
              const routeItem = items.find(item => item.id == id)
              if (routeItem === undefined) return <error404 />
              return <HomeCard {...routeItem} />
            }} />
            <Route component={error404}/>
          </Switch>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  items: state.items,
  error: state.error
})

export const mapDispatchToProps = (dispatch) => ({
  fetchAllItems: (url) => dispatch(fetchAllItems(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
