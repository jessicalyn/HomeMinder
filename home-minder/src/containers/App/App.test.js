import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { mockData } from '../../mockData';
import { shallow } from 'enzyme';
import { fetchAllItems } from '../../thunks/fetchAllItems'

describe('App Component', () => {
  describe('App', () => {
    let wrapper
    let mockfetchAllItems

    const mockItems = mockData

    beforeEach(() => {
      mockfetchAllItems = jest.fn().mockImplementation(() => Promise.resolve({results: mockItems}))
      wrapper = shallow(<App fetchAllItems={mockfetchAllItems} />)
    });

    it('should match the snapshot with the data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it("should invoke the function fetchAllItems on componentDidMount", async () => {
      wrapper.instance().componentDidMount()
      expect(mockfetchAllItems).toBeCalled()
    });

    it("should call fetchAllItems with expected url", async () => {
      const mockUrl = 'http://localhost:3001/api/v1/home'
      wrapper.instance().componentDidMount()
      expect(mockfetchAllItems).toHaveBeenCalledWith(mockUrl)
    })
  });

  describe('mapStateToProps', () => {
    it('should return an object with an items array and an error string', () => {
      const mockState = {
        items: mockData,
        error: "",
        loading: false
      }
      const expected = {
        items: mockData,
        error: ""
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  });

  describe('mapDispatchToProps', () => {
    it.skip('should call dispatch when using fetchAllItems from MDTP', () => {
      const mockDispatch = jest.fn()
      const mockUrl = 'http://localhost:3001/api/v1/items'
      const actionToDispatch = fetchAllItems(mockUrl)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.fetchAllItems(mockUrl)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
