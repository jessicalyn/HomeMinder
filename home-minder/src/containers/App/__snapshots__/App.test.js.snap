// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App should match the snapshot with the data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchAllItems={
      [MockFunction] {
        "calls": Array [
          Array [
            "http://localhost:3001/api/v1/home",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="app-header"
        >
          <h1>
            HomeMinder
          </h1>
          <NavLink
            className="nav-link"
            to="/"
          >
            Home
          </NavLink>
          <NavLink
            className="nav-link"
            to="/audit"
          >
            Audit
          </NavLink>
        </header>,
        <div
          className="app-container"
        >
          <Switch>
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(HomeContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Audit)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/audit"
            />
            <Route
              path="/item/:id"
              render={[Function]}
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              HomeMinder
            </h1>,
            <NavLink
              className="nav-link"
              to="/"
            >
              Home
            </NavLink>,
            <NavLink
              className="nav-link"
              to="/audit"
            >
              Audit
            </NavLink>,
            undefined,
          ],
          "className": "app-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HomeMinder",
            },
            "ref": null,
            "rendered": "HomeMinder",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Home",
              "className": "nav-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Audit",
              "className": "nav-link",
              "to": "/audit",
            },
            "ref": null,
            "rendered": "Audit",
            "type": [Function],
          },
          undefined,
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(HomeContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              exact={true}
              path="/"
            />
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Audit)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                }
              }
              path="/audit"
            />
            <Route
              path="/item/:id"
              render={[Function]}
            />
            <Route
              component={[Function]}
            />
          </Switch>,
          "className": "app-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(HomeContainer)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                exact={true}
                path="/"
              />,
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(Audit)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/audit"
              />,
              <Route
                path="/item/:id"
                render={[Function]}
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(HomeContainer)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(Audit)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
                "path": "/audit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/item/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app-header"
          >
            <h1>
              HomeMinder
            </h1>
            <NavLink
              className="nav-link"
              to="/"
            >
              Home
            </NavLink>
            <NavLink
              className="nav-link"
              to="/audit"
            >
              Audit
            </NavLink>
          </header>,
          <div
            className="app-container"
          >
            <Switch>
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(HomeContainer)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                exact={true}
                path="/"
              />
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(Audit)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/audit"
              />
              <Route
                path="/item/:id"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                HomeMinder
              </h1>,
              <NavLink
                className="nav-link"
                to="/"
              >
                Home
              </NavLink>,
              <NavLink
                className="nav-link"
                to="/audit"
              >
                Audit
              </NavLink>,
              undefined,
            ],
            "className": "app-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HomeMinder",
              },
              "ref": null,
              "rendered": "HomeMinder",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Home",
                "className": "nav-link",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Audit",
                "className": "nav-link",
                "to": "/audit",
              },
              "ref": null,
              "rendered": "Audit",
              "type": [Function],
            },
            undefined,
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(HomeContainer)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                exact={true}
                path="/"
              />
              <Route
                component={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(Audit)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  }
                }
                path="/audit"
              />
              <Route
                path="/item/:id"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>,
            "className": "app-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={
                    Object {
                      "$$typeof": Symbol(react.memo),
                      "WrappedComponent": [Function],
                      "compare": null,
                      "displayName": "Connect(HomeContainer)",
                      "type": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  exact={true}
                  path="/"
                />,
                <Route
                  component={
                    Object {
                      "$$typeof": Symbol(react.memo),
                      "WrappedComponent": [Function],
                      "compare": null,
                      "displayName": "Connect(Audit)",
                      "type": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  path="/audit"
                />,
                <Route
                  path="/item/:id"
                  render={[Function]}
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(HomeContainer)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  },
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": Object {
                    "$$typeof": Symbol(react.memo),
                    "WrappedComponent": [Function],
                    "compare": null,
                    "displayName": "Connect(Audit)",
                    "type": [Function],
                    Symbol(Symbol.hasInstance): [Function],
                  },
                  "path": "/audit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/item/:id",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
